### PR TITLE
[WOE] Implement The End

### PR DESCRIPTION
**Pull request by Susucre**
Tweaked the `NamedPredicate` to return false on empty name, as no Card has "" as its name.

Notably for The End, exiling a Morph should allow to look at hand/libary, but no card can be exiled from there.

### CODE REVIEW

**Comment by Susucre** [comment]
[[The End]]

**Comment by Susucre** [comment]
oops forgot the cost reduction part. will add that shortly.

File: Mage.Sets/src/mage/cards/t/TheEnd.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        if(permanent == null) {
+            return false;
+        }
+        String name = permanent.getName();
+        Player player = game.getPlayer(permanent.getControllerId());
+        if(player == null) {
+            return false;
+        }
+
+        permanent.moveToExile(null, "", source, game);
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/t/TheEnd.java:82]: This seems like an inner method, not to be called from a card.

Normally you'd use player.moveCards or something like.

Comment (Susucre) [reply Mage.Sets/src/mage/cards/t/TheEnd.java:82]: Sure I'll change that.
Is in used in many cards code though
image

**Review by xenohedron — APPROVED** [review]

**Pull request merged**